apps/web/pages/_document.tsx

```diff
@@ -67,7 +67,27 @@ class MyDocument extends Document<Props> {
             nonce={nonce}
             id="newLocale"
             dangerouslySetInnerHTML={{
-              __html: `window.calNewLocale = "${newLocale}";`,
+              __html: `
+              window.calNewLocale = "${newLocale}";
+              (function applyTheme() {
+                  try {
+                  const theme = localStorage.getItem('app-theme');
+                  if (!theme) return;
+
+                  const onReady = () => {
+                    if (document.body) {
+                      document.body.classList.add(theme);
+                    } else {
+                      requestAnimationFrame(onReady);
+                    }
+                  };
+
+                  requestAnimationFrame(onReady);
+                } catch (e) {
+                  console.error('Error applying theme:', e);
+                }
+              })();
+            `,
```

Comment:
Would this not be better living in a useEffect in _app for example? 
```
  useEffect(() => {
    // Apply theme
    const applyTheme = () => {
      try {
        const theme = localStorage.getItem('app-theme');
        if (theme) {
          document.body.classList.add(theme);
        }
      } catch (e) {
        console.error('Error applying theme:', e);
      }
    };

    applyTheme();
  }, []);
``` 

Should also solve a simular effect?

Reply:
In my opinion; in this case it's preferable to have it in _document - as this is a bug that only affects full page reloads and has little to do with the app itself (also I tried your proposed solution and it didn't work for me). But the useEffect would run on every page whereas _document only runs during the very first load.